src/types/resolveDescriptors.ts

```diff
@@ -1947,25 +1947,6 @@ export function resolveDescriptors(ctx: CompilerContext) {
         staticConstants.set(idText(a.name), buildConstantDescription(a));
     }
 
-    // Check for collisions in getter method IDs
```

Comment:
LGTM, just one question: Is there a reason as to why this deleted code was moved to `resolveStatements.ts`?

Reply:
Yep, we need contract constants to work in method IDs too. And the `self` parameter only enters the typing context before we typecheck a function's body in `resolveStatemtens.ts`. This is why the code for type-checking a getter's method id is placed *after* introducing `self` into the context and *before* adding the function parameters.

Reply:
Got it.

Reply:
btw, my guess is that #924 needs to be fixed in the same way